### DISCUSSION TITLE
About: How can I skip the execution of download() and/or process() ?

### DISCUSSION THREAD

**Discussion by Nokimann** [pyg-team/pytorch_geometric#4132] [Q&A] [answered]
In [Creating “Larger” Datasets](https://pytorch-geometric.readthedocs.io/en/latest/notes/create_dataset.html#creating-larger-datasets), to load an exist dataset, we have to re-define Dataset class without override `download()` and `process()`.

I wonder if there is another way without re-defining.

**Comment by rusty1s** [comment, marked as answer]
Can you explain the issue in more detail? In general, downloading and processing are skipped if they are not implemented as part of your `Dataset` class.

**Reply by Nokimann** [reply]
As explained in [Frequently Asked Questions](https://pytorch-geometric.readthedocs.io/en/latest/notes/create_dataset.html#frequently-asked-questions), we have not overridden `download()` and `process()`. To process some personal data, we need to make `class` with overriding `download()` and `process()`. That means we have to either re-define the `class` or define another `class` to load data.

I posted in [issues](https://github.com/pyg-team/pytorch_geometric/issues/4131#issuecomment-1050725201). But, it is solved by [Frequently Asked Questions](https://pytorch-geometric.readthedocs.io/en/latest/notes/create_dataset.html#frequently-asked-questions).

**Reply by rusty1s** [reply]
So is this issue resolved? Otherwise, I am still unsure about what you are trying to achieve. Can you give a small example/code snippet? Happy to help you out!

**Reply by Nokimann** [reply]
Let's say we have two custom datasets: one from `InMemoryDataset` and another from `Dataset`.

Let's use the two datasets **twice**. In general,

1. `InMemoryDataset`
1st time usage: `download()` -> `process()` -> loaded data
2nd time re-usage later: just load processed data


3. `Dataset`
1st time usage: `download()` -> `process()` -> loaded data
2nd time re-usage later: `download()` -> `process()` -> loaded data

So, the step  `download()` -> `process()` is redundant at the 2nd load in the case `Dataset`. To remove the redundancy, we have to redefine the custom datasets before the 2nd usage.

**Reply by rusty1s** [reply]
I don't think this is true. The pipeline is the same for both `InMemoryDataset` and `Dataset`. As long as you define the `raw_file_names` and `processed_file_names` properties in a dataset, downloading and processing will be skipped.

**Reply by Nokimann** [reply]
I found my fault. The names of saved files are different from the names defined in `processed_file_names`. So, it repeatedly generates the dataset from scratch.

**Reply by rusty1s** [reply]
Ah, yes. That makes sense. Glad that this is now resolved :)
